extras/create_snapshots_reports_scorecard.py

```diff
@@ -215,49 +225,147 @@ def create_weekly_snapshots(db, cyhy_db_section):
+            successful_snapshots.append(org_id)
+            if org_descendants and not use_only_existing_snapshots:
+                logging.info(
+                    " - Includes successful descendant snapshot(s): %s", org_descendants
+                )
+                successful_snapshots.extend(org_descendants)
+    else:
+        logging.error("Unsuccessful snapshot: %s", org_id)
+        with fs_lock:
+            failed_snapshots.append(org_id)
+            if org_descendants and not use_only_existing_snapshots:
+                logging.error(
+                    " - Unsuccessful descendant snapshot(s): %s", org_descendants,
+                )
+                failed_snapshots.extend(org_descendants)
+        logging.error("Stderr failure detail: %s %s", data, err)
+    return snapshot_process.returncode
+
+
+def create_snapshots(org_list, db, cyhy_db_section):
+    """Create a snapshot for each organization in a list."""
+    for org_id in org_list:
+        logging.info(
+            "[%s] Starting snapshot for: %s", threading.current_thread().name, org_id
         )
-        data, err = output.communicate("yes")
-        snap_time = time.time() - snap_time
-        longest_snaps.append((i, snap_time))
-        return_code = output.returncode
+        create_snapshot(db, cyhy_db_section, org_id, use_only_existing_snapshots=False)
 
-        if return_code == 0:
-            successful_snaps.append(i)
-            # TODO: If the org has children & their requestdoc has CYHY, add to descendant org list
-            successful_descendant_snaps += db.RequestDoc.get_all_descendants(i)
-            logging.info(
-                "Added to successful snaps: %s (%.2f s)", i, round(snap_time, 2)
+
+def generate_weekly_snapshots(db, cyhy_db_section):
+    """Generate all snapshots needed in order to generate the CyHy reports."""
+    start_time = time.time()
+
+    logging.info("Building list of reports to generate...")
+    reports_to_generate = create_list_of_reports_to_generate(db)
+
+    logging.info("Building list of snapshots to generate...")
+    snapshots_to_generate = create_list_of_snapshots_to_generate(
+        db, reports_to_generate
+    )
+
+    # List to keep track of our snapshot creation threads
+    snapshot_threads = list()
+
+    # Lists of orgs for each snapshot thread to process
+    snapshots_to_generate = list(
+        make_list_chunks(snapshots_to_generate, SNAPSHOT_THREADS)
+    )
+
+    # Start up the threads to create snapshots
+    for orgs in snapshots_to_generate:
+        try:
+            snapshot_thread = threading.Thread(
+                target=create_snapshots, args=(orgs, db, cyhy_db_section),
             )
-        else:
-            failure_snaps.append(i)
-            logging.info("Added to failed snaps: %s", i)
-            logging.info("Stderr failure detail: %s%s", data, err)
+            snapshot_threads.append(snapshot_thread)
+            snapshot_thread.start()
+        except:
```

Comment:
We shouldn't catch `BaseException`.

```suggestion
        except Exception:
```